automation/workflow/hook.go

```diff
@@ -0,0 +1,24 @@
+package workflow
+
+import "github.com/qri-io/qri/event"
+
+// "encoding/json"
+
+// HookType is the type of hook
+type HookType string
+
+// A Hook determines under what circumstances its `hook.Event()` should be
+// emitted, and what the event payload should be.
+type Hook interface {
+	// json.Marshaler
+	// json.Unmarshaler
+	ID() ID
+	IsEnabled() bool
+	Enable() error
+	Disable() error
+	Type() HookType
+	ToMap() map[string]interface{}
+	Advance() error
+	Event() event.Type
+	Payload() interface{}
```

Comment:
Is there any reason not to combine these?

```go
Event() (t event.Type, payload interface{})
```

Reply:
None, thank you, this is great!